pkg/kubelet/container/runtime.go

```diff
@@ -81,4 +81,34 @@ type Container struct {
 	// Hash of the container, used for comparison. Optional for containers
 	// not managed by kubelet.
 	Hash uint64
+	// The timestamp of the creation time of the container.
+	// TODO(yifan): Consider to move it to api.ContainerStatus.
+	Created int64
+}
+
+type Pods []*Pod
+
+// FindPodByID returns a pod in the pod list by UID. It returns nil
+// if not found.
+// TODO(yifan): Use a map?
+func (p Pods) FindPodByID(podUID types.UID) Pod {
+	for i := range p {
+		if p[i].ID == podUID {
+			return *p[i]
+		}
+	}
+	return Pod{}
+}
+
+// FindContainerByName returns a container in the pod with the given name.
+// Note that when there are multiple containers with the same name, then only
```

Comment:
nit: We can just say that the first match will be returned, the user can have the data sorted in whatever order they'd like.